### ISSUE TITLE
IDE0055 triggered over collection expressions (but not array initializers)

### ISSUE THREAD

**Issue by Sergio0694** [dotnet/roslyn#70951]
**Version Used**:  4.8.0-7.23558.1 (e0917286) (from VS 17.8 retail, .NET 8 GA)

**Steps to Reproduce**:

1. Paste the following code:

```csharp
ReadOnlySpan<byte> data =
[
    0x8D, 0xD4, 0xE6, 0x02,
    0x92, 0xB8,
    0xBC, 0x4F,
    0xAA, 0x54,
    0x11,
    0x92,
    0x03,
    0xBA,
    0xB8,
    0x02
];
```

2. Enable all style analyzers

**Diagnostic Id**: IDE0055

**Expected Behavior**:

This should not result in a warning

**Actual Behavior**:

I get an IDE0055 saying I need to "fix formatting":

![image](https://github.com/dotnet/roslyn/assets/10199417/644415d2-5de6-40dc-87e9-325543fee3d4)

This **only** happens with a collection expression. Using `new byte[] { ... }` doesn't show any warnings.

cc. @CyrusNajmabadi

**Comment by DoctorKrolic** [comment]
This is no longer reproducible for latest public preview of VS (17.9 P3)

**Comment by CyrusNajmabadi** [comment]
Thanks!

**Comment by Sergio0694** [comment]
Shouldn't this be closed as fixed? 🤔
